Tensile/SolutionStructs.py

```diff
@@ -44,12 +46,277 @@ def pvar(state, field):
+        self.registerA( [(nidx,Fbs.Batch,ndim)] + spatialDims + filterDims + [(cidx,Fbs.Sum,cdim)] )
+      elif self.tensorAFormat in ("CNHW", "CNDHW"):
+        self.registerA( [(cidx,Fbs.Sum,cdim), (nidx,Fbs.Batch,ndim)] + spatialDims + filterDims )
+      else:
+        raise RuntimeError ("unknown activation format '%s'"%self.tensorAFormat)
+
+      problemTypeOut["NumIndicesC"] = 2+len(spatialDims)
+
+      transposeCK =  convolutionType=="ConvolutionBackwardData"
+      ndim.strideB = 0
+      if self.tensorBFormat == "KCYX" and not transposeCK or\
+         self.tensorBFormat == "CKYX" and transposeCK:
+        self.registerB( [(nidx,Fbs.Batch,ndim), (kidx,Fbs.Free,kdim), (cidx,Fbs.Sum,cdim)] + filterDims )
+      elif self.tensorBFormat == "CKYX" and not transposeCK or\
+           self.tensorBFormat == "KCYX" and transposeCK:
+        self.registerB( [(nidx,Fbs.Batch,ndim), (cidx,Fbs.Sum, cdim), (kidx,Fbs.Free,kdim)] + filterDims )
+      elif self.tensorBFormat == "CYXK" and not transposeCK or\
+           self.tensorBFormat == "KYXC" and transposeCK:
+        self.registerB( [(nidx,Fbs.Batch,ndim), (cidx,Fbs.Sum, cdim)] + filterDims + [(kidx,Fbs.Free,kdim)] )
+      elif self.tensorBFormat == "KYXC" and not transposeCK or\
+           self.tensorBFormat == "CYXK" and transposeCK:
+        self.registerB( [(nidx,Fbs.Batch,ndim), (kidx,Fbs.Free, kdim)] + filterDims + [(cidx,Fbs.Sum,cdim)] )
+      else:
+        raise RuntimeError ("unknown weight format '%s'"%self.tensorBFormat)
+
+    if convolutionType=="ConvolutionBackwardWeights":
+      if self.tensorAFormat in ("NCHW", "NCDHW"):
+        self.registerA( [(nidx,Fbs.Batch,ndim), (cidx,Fbs.Sum,cdim)] + spatialDims + filterDims) # TODO, check)
+        self.registerB( [(nidx,Fbs.Batch,ndim), (cidx,Fbs.Sum,cdim)] + spatialDims + filterDims) # TODO, check?
+      else:
+        raise RuntimeError ("unknown activation format '%s'"%self.tensorAFormat)
+      problemTypeOut["NumIndicesC"] = 2
+
+    problemTypeOut["IndexAssignmentsA"] = [x[0] for x in self.indexA[::-1]]
+    problemTypeOut["IndexAssignmentsB"] = [x[0] for x in self.indexB[::-1]]
+
+
+    problemTypeOut["SetConstStrideA"]=[]
+    for (idx,fbs,dim) in self.indexA:
+      if dim.strideA != -1:
+        problemTypeOut["SetConstStrideA"].append([idx,dim.strideA])
+
+    problemTypeOut["SetConstStrideB"]=[]
+    for (idx,fbs,dim) in self.indexB:
+      if dim.strideB != -1:
+        problemTypeOut["SetConstStrideB"].append([idx,dim.strideB])
+
+    if 0 in [x[0] for x in problemTypeOut["SetConstStrideA"]] or \
+       0 in [x[0] for x in problemTypeOut["SetConstStrideB"]]:
+     print ("warning: need UseInitialStrides")
+     #problemTypeOut["UseInitialStrides"] = 1
+   #else:
+     #problemTypeOut["UseInitialStrides"] = 0
+
+
+    #self.printUsage(problemTypeOut)
+
+  # registerA and registerB:
+  # Provide a list of indices in convolution order - these will be reversed when assigned to IndexAssignmentsAB
```

Comment:
docstrings are more useful than preceding comments - they can be queried and used to generate docs.

Also, what do these functions do? It sounds like they provide a list of indices, not that the caller is providing it.

Reply:
added comments, try to explain what they do.